stonesoup/platform/base.py

```diff
@@ -1,37 +1,223 @@
+                                    "set, it will default to ``[m+1 for m in position_mapping]``")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.position_mapping = coerce_to_valid_mapping(self.position_mapping)
+        if self.velocity_mapping is None:
+            self.velocity_mapping = self.position_mapping + 1
+
+    @property
+    def state_vector(self) -> StateVector:
+        """Convenience property to return the state vector of the state."""
+        return self.state.state_vector
+
+    @property
+    def position(self) -> StateVector:
+        """Return the position of the platform.
+
+        Extracted from the state vector of the platform using the platform's
+        :attr:`position_mapping`. This property is settable for fixed platforms, but not for
+        movable ones, where the position must be set by moving the model with a transition model.
+        """
+        return self.state_vector[self.position_mapping]
+
+    @position.setter
+    def position(self, value: StateVector) -> None:
+        self._set_position(value)
+
+    @property
+    def ndim(self) -> int:
+        """Convenience property to return the number of dimensions in which the platform operates.
+
+        Given by the length of the :attr:`position_mapping`
+        """
+        return len(self.position_mapping)
+
+    @property
+    @abstractmethod
+    def orientation(self) -> StateVector:
+        """Return the orientation of the platform.
+
+        Implementation is case dependent and left to the Fixed/Moving subclasses
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def velocity(self) -> StateVector:
+        """Return the velocity of the platform.
+
+        Implementation is case dependent and left to the Fixed/Moving subclasses
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_moving(self) -> bool:
+        """Return the ``True`` if the platform is moving, ``False`` otherwise.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def move(self, timestamp: datetime.datetime, **kwargs) -> None:
+        """Update the platform position using the :attr:`transition_model`.
+
+        Parameters
+        ----------
+        timestamp: :class:`datetime.datetime`, optional
+            A timestamp signifying when the end of the maneuver \
+            (the default is ``None``)
+
+        Notes
+        -----
+        This methods updates the value of :attr:`position`.
+
+        Any provided ``kwargs`` are forwarded to the :attr:`transition_model`.
+
+        If :attr:`transition_model` or ``timestamp`` is ``None``, the method has
+        no effect, but will return successfully.
+
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def _set_position(self, value: StateVector) -> None:
+        raise NotImplementedError
+
+
+class FixedPlatform(Platform):
```

Comment:
I'd be tempted to make this the base `Platform` class. Maybe even just having MovingPlatform as the base class, with transitional model optional?

Reply:
I just had to look back at the code to find out why this is. The reason is that `orientation` is a `Property` in `FixedPlatform` but a `property` (note P/p) in `MovingPlatform`. I think overriding `Property`s with `property`s was problematic, but I can't remember for certain. It may just be that I thought it unintuitive. 

Moving platform with optional transition model could work, but there would be lots of checks for `self.transition_model is None` as they behave differently in e.g. velocity, orientation.